test/integration/test_timeout.py

```diff
@@ -0,0 +1,214 @@
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import time
+import unittest
+
+import requests
+from unittest.mock import Mock
+from requests import Response
+from urllib3 import Timeout
+
+from ibmcloudant.cloudant_v1 import CloudantV1
+from ibm_cloud_sdk_core.authenticators import BasicAuthenticator, IAMAuthenticator, NoAuthAuthenticator
+from ibmcloudant.couchdb_session_authenticator import CouchDbSessionAuthenticator
+
+DEFAULT_TIMEOUT = 150 # 2.5m (=150s)
+CUSTOM_TIMEOUT = 10   # 10s
+
+class TestTimeout(unittest.TestCase):
+
+# ** Helpers
+# *************
+
+    _timeout_mock_response = Response()
+    _timeout_mock_response.status_code=200
+    _timeout_mock_response._content = '{"foo":"bar"}'.encode()
+    _timeout_mock_response.cookies.set("AuthSession", "bar", expires=time.time()+120)
+    _custom_timeout={'timeout':CUSTOM_TIMEOUT}
+
+    def _set_no_timeout(self, new_value:dict):
+        new_value['timeout']
+
+    # Every test case tests an authenticator and its default timeout value at first,
+    # in the second iteration a custom overwrite possibility is checked.
+    def _defineTestCases(self):
+        return [
+            {
+                'assert_func': self._assert_default_timeout_setting,
+                'set_timeout': self._set_no_timeout
+            },
+            {
+                'assert_func': self._assert_custom_timeout_setting,
+                'set_timeout': self.cloudant.set_http_config
+            }
+        ]
+
+    def _mock_out_cloudant_request(self, _mock_response=_timeout_mock_response):
+        self.cloudant.http_client.request = Mock(return_value=_mock_response)
+
+    def _get_authenticate_arguments(self):
+        requests.request.assert_called_once()
+        return requests.request.mock_calls[0]
+
+    def _get_request_arguments(self, counter):
+        self.assertEqual(self.cloudant.http_client.request.call_count, counter+1)
+        return self.cloudant.http_client.request.mock_calls[counter]
+
+    def _assert_default_timeout_setting(self, call_args):
+        _, _, kwargs = call_args
+        self.assertTrue(isinstance(kwargs['timeout'], Timeout))
+        self.assertEqual(kwargs['timeout'].read_timeout, DEFAULT_TIMEOUT)
+
+    def _assert_custom_timeout_setting(self, call_args):
+        _, _, kwargs = call_args
+        self.assertEqual(kwargs['timeout'], CUSTOM_TIMEOUT)
+
+
+# ** CloudantV1
+# *************
+# Check every authenticator type
+    def test_timeout_cloudantv1_noauth(self):
+        noAuth_auth = NoAuthAuthenticator()
+        self.cloudant = CloudantV1(
+            authenticator=noAuth_auth
+        )
+
+        self._mock_out_cloudant_request()
+
+        testcases = self._defineTestCases()
+
+        for i, tc in enumerate(testcases):
```

Comment:
We should rename `i` to a meaningful variable name e.g. `count`

Reply:
The `i` would refer to `index` or `iteration`, this is a common abbreviation. `count` implies something is counted. I would go with `idx` or `index` rather. WDYT?

Reply:
Changed in 340cf87

Reply:
Yea that's fine.  I reason I suggested `count` is because of the `counter` argument in `_get_request_arguments` func signature:
```
def _get_request_arguments(self, counter)
```

Reply:
Oh, I see now!
I've renamed `idx` to `tc_num`. Changed here: 5131d84